Assets/Mona/QualityAssurance/QualityAssurance.Codes.cs

```diff
@@ -0,0 +1,35 @@
+public static class MonaErrorCodes
+{
+    // Missing Layer Errors
+    public static string MISSING_SPACE_LAYER = "space.missing-layer";
+    public static string MISSING_ARTIFACT_LAYER = "artifact.missing-layer";
+    public static string MISSING_PORTAL_LAYER = "portal.missing-layer";
+
+    // Missing Scene Errors
+    public static string MISSING_SPACE_SCENE = "space.missing-scene";
+    public static string MISSING_PORTAL_SCENE = "portal.missing-scene";
+    public static string MISSING_ARTIFACT_SCENE = "artifact.missing-scene";
+
+    // Multiple Root Object Errors
+    public static string MULTIPLE_SPACE_ROOTS = "space.multiple-roots";
+    public static string MULTIPLE_ARTIFACT_ROOTS = "artifact.multiple-roots";
+    public static string MULTIPLE_PORTAL_ROOTS = "portal.multiple-roots";
+
+    // Object Placement errors
+    public static string BAD_ARTIFACT_PLACEMENT = "artifact.bad-placement";
+    public static string BAD_CANVAS_PLACEMENT = "canvas.bad-placement";
+    public static string BAD_PORTAL_PLACEMENT = "portal.bad-placement";
+
+    // Duplicate Name Errors
+    public static string DUPLICATE_ARTIFACT_NAME = "artifact.duplicate-name";
+    public static string DUPLICATE_CANVAS_NAME = "canvas.duplicate-name";
+    public static string DUPLICATE_PORTAL_NAME = "portal.duplicate-name";
+
+    // Layer Content Errors
+    public static string BAD_PORTAL_LAYER_CONTENTS = "portal.layer-contents";
+
+    // Collider Errors
+    public static string BAD_PORTAL_COLLIDER = "portal.bad-collider";
+    public static string BAD_ARTIFACT_COLLIDER = "artifact.bad-collider";
+    public static string BAD_CANVAS_COLLIDER = "canvas.bad-collider";
```

Comment:
These error codes are written in localization format. You might want to consider either writing out the actual text here and/or including a code-to-text converter. Check out the localization files in Zodiac for reference. https://github.com/monaverse/Zodiac/blob/a96798591ee4c30888fc578812de600ade71ad9d/locales/en.js#L18

Reply:
Actually, do we have this JS file exposed?
In that case, we could actually just download the Locale file from Mona into the editor and keep it automatically updated

Reply:
No, not exposed unfortunately. Zodiac will be multi-lang while this will likely only be English, so that might be overkill for now. I think some code-to-text converter would probably be best